actions/image/crop.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Image crop action
+ */
+
+$guid = get_input('guid');
+$entity = get_entity($guid);
+
+if (!$entity || !($entity instanceof ElggEntity) || !$entity->canEdit()) {
+	register_error(elgg_echo('image:crop:fail'));
+	forward(REFERER);
+}
+
+$x1 = (int) get_input('x1', 0);
+$y1 = (int) get_input('y1', 0);
+$x2 = (int) get_input('x2', 0);
+$y2 = (int) get_input('y2', 0);
+
+$filehandler = new ElggFile();
+$filehandler->owner_guid = $entity->getGUID();
+$filehandler->setFilename("icon/master.jpg");
```

Comment:
https://github.com/Elgg/Elgg/blob/master/mod/groups/actions/groups/edit.php#L205
https://github.com/Elgg/Elgg/blob/master/actions/avatar/upload.php#L32
Not to mention the madness that happens with files: https://github.com/Elgg/Elgg/blob/master/mod/file/actions/file/upload.php#L113